[CI Visibility] - Add new section for BenchmarkDotnet support. (#17016)
DataDog/documentation: content/en/continuous_integration/tests/dotnet.md

To create filters or `group by` fields for these tags, you must first create facets. For more information about adding tags, see the [Adding Tags][7] section of the .NET custom instrumentation documentation.

### Instrumenting BenchmarkDotNet tests

To instrument your benchmark tests you need to:

> 1. Add the [`Datadog.Trace.BenchmarkDotNet` NuGet package][12] to your project (eg. using `dotnet add package Datadog.Trace.BenchmarkDotNet`).

Reviewer: ```suggestion
1. Add the [`Datadog.Trace.BenchmarkDotNet` NuGet package][12] to your project (for example, using `dotnet add package Datadog.Trace.BenchmarkDotNet`).
```

2. Configure your project to use the `Datadog.Trace.BenchmarkDotNet` exporter using the `DatadogDiagnoser` attribute or the `WithDatadog()` extension method. For example:

{{< tabs >}}

> {{% tab "Using the Attribute" %}}

Reviewer 2: ```suggestion
{{% tab "Using the `[DatadogDiagnoser]` Attribute" %}}
```

{{< code-block lang="csharp" >}}
using BenchmarkDotNet.Attributes;
using Datadog.Trace.BenchmarkDotNet;

[DatadogDiagnoser]
[MemoryDiagnoser]
public class OperationBenchmark
{
    [Benchmark]
    public void Operation()
    {
        // ...
    }
}
{{< /code-block >}}
{{% /tab %}}

{{% tab "Using the Configuration" %}}
{{< code-block lang="csharp" >}}
using BenchmarkDotNet.Configs;
using BenchmarkDotNet.Running;
using Datadog.Trace.BenchmarkDotNet;

var config = DefaultConfig.Instance
              .WithDatadog();

BenchmarkRunner.Run<OperationBenchmark>(config);
{{< /code-block >}}
{{% /tab %}}

{{< /tabs >}}

> 3. Configure the reporting method as [explained above][13]

Reviewer: ```suggestion
3. [Configure the reporting method][13].
```
